gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/test/java/org/wso2/carbon/connector/integration/test/common/ConnectorIntegrationUtil.java

```diff
@@ -0,0 +1,426 @@
+
+package org.wso2.carbon.connector.integration.test.common;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.util.AXIOMUtil;
+import org.apache.axiom.soap.SOAP12Constants;
+import org.apache.axiom.soap.SOAPEnvelope;
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.Constants;
+import org.apache.axis2.addressing.EndpointReference;
+import org.apache.axis2.client.OperationClient;
+import org.apache.axis2.client.Options;
+import org.apache.axis2.client.ServiceClient;
+import org.apache.axis2.context.MessageContext;
+import org.apache.axis2.transport.TransportUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.carbon.automation.core.ProductConstant;
+import org.wso2.carbon.mediation.library.stub.upload.MediationLibraryUploaderStub;
+import org.wso2.carbon.mediation.library.stub.upload.types.carbon.LibraryFileItem;
+import javax.activation.DataHandler;
+import javax.xml.stream.XMLStreamException;
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.rmi.RemoteException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+public class ConnectorIntegrationUtil {
+
+    public static final String ESB_CONFIG_LOCATION = "artifacts" + File.separator + "ESB" + File.separator + "config";
+    private static final Log log = LogFactory.getLog(ConnectorIntegrationUtil.class);
+    public static void uploadConnector(String repoLocation, MediationLibraryUploaderStub mediationLibUploadStub,
+                                       String strFileName) throws MalformedURLException, RemoteException {
+        List<LibraryFileItem> uploadLibraryInfoList = new ArrayList<LibraryFileItem>();
+        LibraryFileItem uploadedFileItem = new LibraryFileItem();
+        uploadedFileItem.setDataHandler(new DataHandler(new URL("file:" + "///" + repoLocation + "/" + strFileName)));
+        uploadedFileItem.setFileName(strFileName);
+        uploadedFileItem.setFileType("zip");
+        uploadLibraryInfoList.add(uploadedFileItem);
+        LibraryFileItem[] uploadServiceTypes = new LibraryFileItem[uploadLibraryInfoList.size()];
+        uploadServiceTypes = uploadLibraryInfoList.toArray(uploadServiceTypes);
+        mediationLibUploadStub.uploadLibrary(uploadServiceTypes);
+    }
+
+    public static int sendRequestToRetriveHeaders(String addUrl, String query) throws IOException, JSONException {
+        String charset = "UTF-8";
```

Comment:
This variable is created in each methods please move to the constant class and reuse.